src/frontends/tensorflow/src/checkpoint_v1_reader.cpp

```diff
@@ -0,0 +1,250 @@
+        FRONT_END_GENERAL_CHECK(sts.ParseFromArray(value.data(), static_cast<int>(value.size())),
+                                "[TensorFlow Frontend] incorrect input checkpoint file or internal error: cannot parse "
+                                "SavedTensorSlices entry");
+        for (const auto& saved_slice_meta : sts.meta().tensor()) {
+            // parse shapes and types for variables
+            VariableInfo var_info;
+            var_info.shard_id = shard_ind;
+            auto variable_name = saved_slice_meta.name();  // original variable name (not encoded)
+            var_info.variable_shape = saved_slice_meta.shape();
+            var_info.variable_type = saved_slice_meta.type();
+            var_info.is_initialized = false;
+
+            // save starts and lenghts of slices for variable name encoding
+            for (const auto& slice : saved_slice_meta.slice()) {
+                // var_info.starts.push_back(slice.extent())
+                for (const auto& extent : slice.extent()) {
+                    var_info.starts.push_back(extent.start());
+                    if (extent.has_length()) {
+                        var_info.lenghts.push_back(extent.length());
+                    } else {
+                        var_info.lenghts.push_back(-1);
+                    }
+                }
+            }
+
+            m_variables_info_map[variable_name] = var_info;
+        }
+        ++shard_ind;
+    }
+}
+
+void CheckpointV1Reader::seek_block(const std::string& target_key,
+                                    const char* block_ptr,
+                                    const uint32_t restarts,
+                                    std::string& value) const {
+    // parsing the next key starts at the end of value, so set value accordingly
+    const char* curr_value_pos = block_ptr;
+    const char* limit = block_ptr + restarts;  // restarts come right after data
+    std::string key = "";
+
+    bool is_found = false;
+    while (true) {
+        FRONT_END_GENERAL_CHECK(
+            curr_value_pos < limit,
+            "[TensorFlow Frontend] incorrect model: no more entries to return, invalid checkpoint file");
+
+        // decode next entry
+        // each entry looks as follows:
+        // | shared (1 byte) | non-shared (1 byte) | value_length (1 byte) | key (non-shared bytes) |
+        // | value (value_length bytes) |
+        uint32_t shared, non_shared, value_length;
+        curr_value_pos = decode_entry(curr_value_pos, limit, shared, non_shared, value_length);
+        FRONT_END_GENERAL_CHECK(curr_value_pos && key.size() >= shared,
+                                "[TensorFlow Frontend] incorrect model: corruption error in checkpoint file");
+
+        key.resize(shared);
+        key.append(curr_value_pos, non_shared);
+        value = std::string(curr_value_pos + non_shared, value_length);
+        curr_value_pos += (non_shared + value_length);
+
+        if (key.compare(target_key) >= 0) {
```

Comment:
Reminder, as we discussed - we may try to cache size/offset of all variables here.
Possible issue: if TF may have a garbage (like unused variables data) it may fail.

Reply:
Unfortunately, this approach does not work. It is triggered on inexistent variable names for which decoding block offset and size fails. That is probably a reason why TF does not implement this this way either.